dotnet/src/SemanticKernel.Skills/Skills.OpenAPI/Extensions/KernelOpenApiExtensions.cs

```diff
@@ -0,0 +1,146 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Text.Json.Nodes;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.SemanticKernel.Diagnostics;
+using Microsoft.SemanticKernel.Orchestration;
+using Microsoft.SemanticKernel.SkillDefinition;
+using Microsoft.SemanticKernel.Skills.OpenAPI.Model;
+using Microsoft.SemanticKernel.Skills.OpenAPI.OpenApi;
+using RestSkills;
+using RestSkills.Authentication;
+
+namespace Microsoft.SemanticKernel.Skills.OpenAPI.Extensions;
+
+/// <summary>
+/// Class for extensions methods for IKernel interface.
+/// </summary>
+public static class KernelOpenApiExtensions
+{
+    public static IDictionary<string, ISKFunction> RegisterOpenApiSkill(this IKernel kernel, Stream documentStream, string skillName)
```

Comment:
We'll want a overload of this that loads directly from a json url, and one from a local file json path (maybe these are the same... both string paths)